crates/next-dev/src/lib.rs

```diff
@@ -140,38 +140,28 @@ impl NextDevServerBuilder {
         self
     }
 
-    pub async fn build(self) -> Result<DevServer> {
-        let start_port = self.port.context("port must be set")?;
-        let host = self.hostname.context("hostname must be set")?;
-
-        // Retry to listen on the different port if the port is already in use.
-        let mut bound_server = None;
-        for retry_count in 0..10 {
-            let current_port = start_port + retry_count;
+    /// Attempts to find an open port to bind.
+    fn find_port(&self, host: IpAddr, port: u16, max_attempts: u16) -> Result<DevServerBuilder> {
+        // max_attempts of 1 means we loop 0 times.
+        let max_attempts = max_attempts - 1;
```

Comment:
Could this argument be "retry" instead? That way we avoid integer underflow altogether. In any case, the semantics of `max_attempts = 0` are not very clear. You could also use https://doc.rust-lang.org/beta/std/num/struct.NonZeroU16.html.